data-avro/src/test/java/com/linkedin/data/avro/TestDataTranslator.java

```diff
@@ -1075,6 +1075,41 @@ public void testDataTranslator() throws IOException
     }
   }
 
+  @Test
+  public void testInfinityAndNan() throws IOException {
+    String schemaText =
+        "{\n" +
+          "  \"type\" : \"record\",\n" +
+          "  \"name\" : \"Foo\",\n" +
+          "  \"fields\" : [\n" +
+          "    { \"name\" : \"doubleRequired\", \"type\" : \"double\" },\n" +
+          "    { \"name\" : \"floatRequired\", \"type\" : \"float\" }\n" +
+          "  ]\n" +
+          "}\n";
+    // First element is the input, second element is the expected output
+    Object[][] inputs = {
+        {
+            "{ \"doubleRequired\" : \"Infinity\", \"floatRequired\" : \"Infinity\"}",
+            Double.POSITIVE_INFINITY
+        }, {
+            "{ \"doubleRequired\" : \"NaN\", \"floatRequired\" : \"NaN\"}",
+            Double.NaN
+        }, {
+            "{ \"doubleRequired\" : \"-Infinity\", \"floatRequired\" : \"-Infinity\"}",
+            Double.NEGATIVE_INFINITY
+        }
+    };
+    RecordDataSchema recordDataSchema = (RecordDataSchema) TestUtil.dataSchemaFromString(schemaText);
+    for (Object[] input : inputs) {
+      DataMap dataMap = TestUtil.dataMapFromString((String) input[0]);
+      Schema avroSchema = SchemaTranslator.dataToAvroSchema(recordDataSchema);
+      GenericRecord avroRecord = DataTranslator.dataMapToGenericRecord(dataMap, recordDataSchema, avroSchema);
+      assertEquals(avroRecord.get("doubleRequired"), input[1]);
```

Comment:
assert float as well.

Reply:
shoot, sorry about that. I merged before I saw your comment. I'll put up another PR with the fix